### DISCUSSION TITLE
Uninstall of client does not remove all configuration files.

### DISCUSSION THREAD

**Discussion by dshizntpdt** [rustdesk/rustdesk#8938] [General]
### Bug Description

When uninstalling rustdesk 1.2.7 from Windows MSI or EXE the files in C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk are kept, keeping old server information if the client is reinstalled.

### How to Reproduce

Uninstall rustdesk and check the folder C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk to see the config folder and files

### Expected Behavior

remove all traces of rustdesk when uninstalling or ask to remove configuration files as well.

### Operating system(s) on local side and remote side

Windows 10, Server 2016

### RustDesk Version(s) on local side and remote side

1.2.7

### Screenshots

NA

**Comment by graci1117** [comment]
How do I uninstall it on windows?

**Comment by thespirit61** [comment]
This would be a nice improvement and somehow better security practise.

**Comment by wprivera** [comment]
Cloned hard drive to new PC. Uninstalled RustDesk client in Windows 11. Rebooted PC. Reinstalled RustDesk. Issued same client ID as previous PC. Uninstalled RustDesk. Scanned registry. Deleted ALL entries related to RustDesk. On SECOND re-install issued same client ID. Deleted Windows RustDesk App folder. Again deleted all registry entries. Rebooted PC.

THIRD re-install attempt - after uninstall of Client software - after deletion of all registry keys - after deletion of RustDesk app folder - after reboot of PC - STILL issued SAME Client ID.  WTF!?!

**Reply by rkolodz** [reply]
Deleting the following has worked for me on Windows:
C:\Users\username\AppData\Roaming\RustDesk\
C:\Users\username\AppData\Local\rustdesk
C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk

**Comment by kd4e** [comment]
I know this is old but I'm observing the same problem on MX Linux (from the GUI) ...

I'm installing the most-recent version of the RustDesk client (June 2025) using the MX Linux Package Installer.
(I've asked on the MX Linux support forum - no one seems to have a solution there.)

A power surge, or something, messed up the admin password for my RustDesk Client install on my MX Linux laptop.
I uninstalled it. Deleted ~/.config/RustDesk.toml, rebooted, then reinstalled.
It fails to prompt me to create an admin password but demands one when I try to change settings.

I systematically went through every instance of anything labeled "rustdesk" and deleted it. The problem persists.

What do I need to do to force RustDesk Client to reinstall - cleanly - so that I can create a new admin password, please?

BTW: I tried loading from terminal and observed this (which may not be related): 
```
$ rustdesk
Cannot load libcuda.so.1
Try setting transparent
Cannot load libcuda.so.1
```

**Reply by wprivera** [reply]
This is what worked for me, in a Windows install. From what I understand, Linux also uses the RustDesk.toml file. I copied the below information from someone else's post:

"Changing manually the ID with a new (another) one in 

C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml does the job, but you have to be sure first it is not already used by someone else and this is almost impossible to find (making a test connection to the ID (in question) to check whether it used is not enough, the remote machine could be offline at that moment) ...

I have found another smarter, easier and safer way - forcing the app to change the ID automatically by changing the following in
C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml for the installed version:

id = '0'
password = ''
salt = ''

You do not need to stop the app at all. After making changes to this file only restart the computer and the work is done ...

id = '0' - is unacceptable value for the app and it takes immediately the actions to search for a new free ID and change the one before ...
password = '' - changes automatically the password, too ... Don't touch, if you want to preserve the password ...
salt = '' - for just in case ...."

This is the official GitHub resource: https://rustdesk.com/docs/en/client/

Perhaps search for the RustDesk.toml file on your Linux install, and apply the same RustDesk.toml edits as described for the Windows install. Don't forget the quotes ( ' ' ) in the file edits. They're important. Hope this helps.

**Reply by kd4e** [reply]
Thank you for your reply.
The error I'm experiencing is not in the Client-to-Cloud-to-Client connection but rather the Client -> Settings Admin Password (see pic).
![rustdesk-password-error-1](https://github.com/user-attachments/assets/7b397e74-7f38-4fe2-9f53-770fa245ed55)

**Reply by kd4e** [reply]
I got an answer - it's a bug in Rustdesk. When used with MX Linux - even though you're inside the Rustdesk GUI - it actually wants your MX Linux password as the Rustdesk Admin password. Sigh. (They should add an info popup.)
